lib/media/streaming_engine.js

```diff
@@ -1743,36 +1744,70 @@ shaka.media.StreamingEngine = class {
     const repId = this.mediaStates_.get(ContentType.VIDEO)
         .stream.originalId.split(',').pop();
     if (repId == null) {
-      shaka.log.warning('getPresentationLatency cannot get video repId');
+      shaka.log.warning('getProducerReferenceTimeInfo cannot get video repId');
       return null;
     }
     if (data.periodIdMap == null) {
-      shaka.log.warning('getPresentationLatency cannot get periodIdMap');
+      shaka.log.warning('getProducerReferenceTimeInfo cannot get periodIdMap');
       return null;
     }
     const periodId = this.getPeriodIdFromPresentationTime(pt, data.periodIdMap);
-    const rv = [];
     const keyPrefix = `${periodId}_${repId}`;
+
+    let mpdPrtInfo = {};
+    let mp4PrftInfo = {
+      wca: [undefined, undefined],
+      pta: [undefined, undefined],
+    };
+
     for (const [key, producerReferenceTime] of data.prtMap.entries()) {
       if (!key.startsWith(keyPrefix)) {
         continue;
       }
       const wca = producerReferenceTime.wca;
       const pta = producerReferenceTime.pta;
       const type = producerReferenceTime.type;
-      const latency = Math.round((wc - wca) - (pt - pta));
-      shaka.log.debug(`computing presentation latency for
+      const wcaPrft = producerReferenceTime.wca_prft;
+      const ptaPrft = producerReferenceTime.pta_prft;
+
+      shaka.log.debug(`Getting ProducerReferenceTimeInfo for
           representation ${repId}, 
           wc=${wc},
           wcOffset=${clockOffset},
-          wca=${wca}, 
-          pt=${pt} 
-          pta=${pta}
-          type=${type}
-          latency=${latency}ms`);
-      rv.push({type, latency});
-    }
-    return rv;
+          wca=${wca},
+          wca from PRFT=${wcaPrft},
+          pt=${pt},
+          pta=${pta},
+          pta from PRFT=${ptaPrft},
+          type=${type}`);
+
+      mpdPrtInfo = {
+        type: 'mpd',
```

Comment:
"type" in PRFT context (https://dashif.org/docs/CR-Low-Latency-Live-r8.pdf) is used to denote encoder vs captured, suggest use "inband": true for prft box source, force for mpd source.